File: prototype/stock-webflux/src/main/java/com/project/stock/advice/GlobalExceptionHandler.java
package com.project.stock.advice;

import com.project.stock.domain.dto.ApiError;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

// GlobalExceptionHandler
@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(IllegalArgumentException.class)
    public ResponseEntity<ApiError> illegalArg(IllegalArgumentException ex) {
        String msg = (ex.getMessage() != null) ? ex.getMessage() : "conflict";
        return ResponseEntity.status(HttpStatus.CONFLICT).body(new ApiError(msg));
    }

    @ExceptionHandler(InvalidCredentialsException.class)
    public ResponseEntity<ApiError> invalidCredentials(InvalidCredentialsException ex) {
        String msg = (ex.getMessage() != null) ? ex.getMessage() : "Invalid Credentials";
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
                .body(new ApiError(msg));
    }
}

File: prototype/lietzsche/src/main/kotlin/com/stxtory/lietzsche/advice/GlobalExceptionHandler.kt
package com.stxtory.lietzsche.advice

import com.stxtory.lietzsche.controller.InvalidCredentials
import org.springframework.http.*
import org.springframework.web.bind.annotation.*

data class ApiError(val message: String)

@RestControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(IllegalArgumentException::class)
    fun illegalArg(ex: IllegalArgumentException) =
        ResponseEntity.status(HttpStatus.CONFLICT).body(ApiError(ex.message ?: "conflict"))
}

@RestControllerAdvice
class ApiExceptionHandler {
    @ExceptionHandler(InvalidCredentials::class)
    fun invalidCredentials(e: InvalidCredentials) =
        ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(ApiError("invalid credentials"))
}